### DISCUSSION TITLE
What's the different bewteen `Disabling Queries` and `enable` properties in `useQuery` options?

### DISCUSSION THREAD

**Discussion by Senbonzakura1234** [trpc/trpc#6216] [Q&A]
I saw [Disabling Queries](https://trpc.io/docs/client/react/disabling-queries) and compare it to [`enable` properties in `useQuery`](https://trpc.io/docs/typedoc/react-query/shared/interfaces/UseTRPCQueryOptions#enabled) options i couldn't see the difference in use case, can someone help me what should i use for when?

**Comment by KATT** [comment]
If you use ....

```tsx
export function MyComponent() {
  const [name, setName] = useState<string | undefined>();
  const result = trpc.getUserByName.useQuery({ name }, { enabled: true });
    return (
      ...
    )
  }
}
```

.. you get a type error if `name` is required on the backend.



If you instead use ...

```ts
import { skipToken } from '@tanstack/react-query';
export function MyComponent() {
  const [name, setName] = useState<string | undefined>();
  const result = trpc.getUserByName.useQuery(name ? { name } : skipToken);
    return (
      ...
    )
  }
}
```

... you get no type error

**Reply by Senbonzakura1234** [reply]
@KATT is there any use case that the [`enable` properties](https://trpc.io/docs/typedoc/react-query/shared/interfaces/UseTRPCQueryOptions#enabled) should be prefer? Cuz with your answer here i see no reason to use it over [Disabling Queries](https://trpc.io/docs/client/react/disabling-queries)
